### PR TITLE
CD-360684 resource limitation

### PR DESCRIPTION
**Pull request by ysedliarou**
https://coupadev.atlassian.net/browse/CD-360684

review:
- [ ] @johnny-lai

## Generated Reviewers

<details>
  <summary><em>:warning: Do not modify anything in this section! :warning:</em></summary>
  <em>The content in this section is managed automatically, and any manual changes to it will be ignored and overwritten the next time the code review status is refreshed! View <a href="https://cody.coupa.engineering/repos/coupa/resque/pull/26">this PR</a> on Cody to see the current code review status.</em>
</details>

### CODE REVIEW

File: lib/resque.rb [hunk 1]
```diff
@@ -223,6 +223,13 @@ def redis_id
   DEFAULT_PRUNE_INTERVAL = DEFAULT_HEARTBEAT_INTERVAL * 5
   DEFAULT_NOISY_NEIGHBOURS_PROTECTION = false
   DEFAULT_DISABLED_HOOKS_LOGGING = [:before_fetch_job]
+  DEFAULT_ON_WAIT_INTERVAL = 5 * 60
+
+  attr_accessor :on_wait_interval
```
Comment (johnny-lai) [inline lib/resque.rb:228]: Maybe `waitpid_interval`?
The `on_` prefix makes me think it is a callback

File: lib/resque/worker.rb [hunk 2]
```diff
@@ -245,12 +246,24 @@ def work(interval = 5.0, &block)
...
-              _, status = Process.waitpid2(@child)
+              pid = @child
+              on_wait_time = Time.now.utc
+              run_hook :before_waitpid, job, pid
+              while status.nil?
+                _, status = Process.waitpid2(@child, Process::WNOHANG)
+                if status.nil?
+                  now = Time.now.utc
+                  if now > on_wait_time + Resque.on_wait_interval.seconds
+                    on_wait_time = now
```
Comment (johnny-lai) [inline lib/resque/worker.rb:257]: Minor: If you used `on_wait_time = now + Resque.on_wait_interval.seconds`, you will do less additions

File: lib/resque.rb [hunk 3]
```diff
@@ -223,6 +223,13 @@ def redis_id
...
+
+  def on_wait_interval
+    @on_wait_interval || DEFAULT_ON_WAIT_INTERVAL
+  end
```
Comment (johnny-lai) [inline lib/resque.rb:232]: Add a test?

File: lib/resque.rb (hunk 1 above)
Comment (ysedliarou) [reply lib/resque.rb:228]: done.

File: lib/resque.rb (hunk 3 above)
Comment (ysedliarou) [reply lib/resque.rb:232]: done.

**Review by johnny-lai — APPROVED** [review]

**Comment by johnny-lai** [comment]
cody rebuild

**Review by johnny-lai — APPROVED** [review]

**Review by johnny-lai — APPROVED** [review]

**Comment by ysedliarou** [comment]
[lockit v5.0.2] merged via Lockit CLI on 2023-07-11 09:31:16.404924 -0700 PDT m=+7.412287884

**Pull request merged**